src/test/java/org/broadinstitute/hellbender/tools/copynumber/CreateBinningIntervalsIntegrationTest.java

```diff
@@ -0,0 +1,221 @@
+        binsExpected.add(new Interval(contig, 2500, 12499));
+        binsExpected.add(new Interval(contig, 12500, 20500));
+
+        // compare the file contents to the expected result
+        Assert.assertEquals(binsFromFile, binsExpected);
+    }
+
+
+    // Test for overlapping intervals
+    @Test
+    public void simpleIntervalTest2() {
+        final int widthOfBins = 10000;
+        final int padding = 500;
+        final File outputFile = createTempFile("GATK-create-binning-intervals-test", ".tmp");
+        final String[] args = {
+                "-L", "20:3000-20000",
+                "-L", "20:200-2100",
+                "-R", REFERENCE.getAbsolutePath(),
+                "-" + CreateBinningIntervals.WIDTH_OF_BINS_SHORT_NAME, Integer.toString(widthOfBins),
+                "-" + CreateBinningIntervals.PADDING_SHORT_NAME, Integer.toString(padding),
+                "-O", outputFile.getAbsolutePath()
+        };
+        runCommandLine(args);
+
+        // read the contents of the output file and print them out
+        // printOutputFile(outputFile);
+
+        // read the bins from the output file
+        IntervalList binsFromFile = IntervalList.fromFile(outputFile);
+
+        // we expect the following result
+        IntervalList binsExpected = new IntervalList(binsFromFile.getHeader().getSequenceDictionary());
+        String contig = binsFromFile.iterator().next().getContig();
+        binsExpected.add(new Interval(contig, 1, 10000));
+        binsExpected.add(new Interval(contig, 10001, 20000));
+        binsExpected.add(new Interval(contig, 20001, 20500));
+
+        // compare the file contents to the expected result
+        Assert.assertEquals(binsFromFile, binsExpected);
+    }
+
+
+    // Test for intervals reaching the ends of the reads
+    @Test
+    public void simpleIntervalTest3() {
+        final int widthOfBins = 10000;
+        final int padding = 500;
+        final File outputFile = createTempFile("GATK-create-binning-intervals-test", ".tmp");
+        final String[] args = {
+                "-L", "20:3000-20000",
+                "-L", "20:63025220-63025520",
+                "-R", REFERENCE.getAbsolutePath(),
+                "-" + CreateBinningIntervals.WIDTH_OF_BINS_SHORT_NAME, Integer.toString(widthOfBins),
+                "-" + CreateBinningIntervals.PADDING_SHORT_NAME, Integer.toString(padding),
+                "-O", outputFile.getAbsolutePath()
+        };
+        runCommandLine(args);
+
+        // read the contents of the output file and print them out
+        // printOutputFile(outputFile);
+
+        // read the bins from the output file
+        IntervalList binsFromFile = IntervalList.fromFile(outputFile);
+
+        // we expect the following result
+        IntervalList binsExpected = new IntervalList(binsFromFile.getHeader().getSequenceDictionary());
+        String contig = binsFromFile.iterator().next().getContig();
+        binsExpected.add(new Interval(contig, 2500, 12499));
```

Comment:
You might want to not get the contig from the file, since you want to make sure that the contig that you wrote to the test file was correct in the first place

Reply:
Oops, didn't see that @asmirnov239 already caught this.

Reply:
Thanks, I corrected it.